Add i18next integration

Josip-Ledic opened the pull request:
### Summary

- [x] add core functionality based on the following syntax: `t('someKey', "Default value in English co-located like this")`
- [x] works with SSR setup
- [x] detects browser language automatically
- [x] sample translations in German
- [x] added documentation to README
- [x] cleaned up code

Josip-Ledic committed:
Add i18next internationalization

- Install i18next and react-i18next with full SSR support
- Configure client-side i18n with browser language detection
- Configure server-side i18n with filesystem backend
- Add improved German translations (idiomatic business German)
- Integrate with Vite via i18next-loader plugin
- Add i18next middleware to Express server
- Update server-side entry to serialize translations for hydration
- Update client-side entry to hydrate with server state
- Translate all Welcome page content with defaultValue pattern
- Document i18n usage and architecture in CLAUDE.md

English text stays inline in components as defaultValue.
Other languages defined in src/locales/*.json files.
Browser automatically detects user's preferred language.
No language flicker due to proper SSR hydration.

Josip-Ledic committed:
feat(i18n): add missing translations in german

Josip-Ledic commented:
I have read the DCO document and I hereby sign the DCO.

Josip-Ledic commented on src/entry-client.jsx:
```diff
@@ -9,18 +9,37 @@
+import i18n from './i18n.client.js';
+
+// Hydrate i18n with server state to prevent flicker
+const { initialI18nStore, initialLanguage } =
```
Without this we get "Willkommen" -> "Welcome" -> "Willkommen" because server first gets the language from the headers, returns German, then the client side is initializing its json language file and because this is async it shows English for a split second and this causes flicker. Happy to do it in a more elegant way but this is what Bob suggested after a lot of back and forth 😄

Josip-Ledic commented on src/i18n.server.js:
```diff
@@ -0,0 +1,46 @@
+    preload: ['en', 'de'], // Preload all languages on server
+
+    // Important for SSR
+    useSuspense: false,
```
ReactDOMServer does not yet support Suspense

rodet replied:
Please add this in the comments. This needs to be an easy to find information for users.

Josip-Ledic committed:
fix language detection

Josip-Ledic committed:
docs(i18n): add initial, minimal description about i18n to readme

Josip-Ledic committed:
update package lock

Josip-Ledic committed:
add missing German translations

Josip-Ledic committed:
Merge remote-tracking branch 'origin/main' into feature/47-i18next

Josip-Ledic committed:
fresh package lock without artifactory usage

rodet commented on vite.config.js:
```diff
@@ -7,10 +7,17 @@
 // https://vite.dev/config/
 export default defineConfig({
-  plugins: [react()],
+  plugins: [
```
Move this into a code comment please.

rodet reviewed (changes requested):
Overall I think this works great, good job @Josip-Ledic . 

Two things to change before we release this:
- Update the copyright years for each file here
- Move a bunch of your PR explanation comments into the code so those don't get lost. They are valuable piece of information that should be in the code.

rodet committed:
chore: upgrade the package version

i18n is a significant feature that warrants this upgrade

Josip-Ledic committed:
Merge remote-tracking branch 'origin/main' into feature/47-i18next

Josip-Ledic committed:
feat(i18n): address PR comments

Josip-Ledic committed:
feat(i18n): fix copyright headers, improve comments, bump package version

rodet committed:
feat(i18n): add dashboard internationalization with German translations

rodet committed:
chore: update package-lock.json after npm install

rodet committed:
chore: merge main branch and resolve conflicts

- Update to eslint-plugin-import-x
- Keep i18n translations in all components
- Adopt new lint-staged configuration with copyright checking
- Maintain improved layout structure from main branch

rodet reviewed (approved):
Let's merge this one - finally!
